### ISSUE TITLE
Clicking the search button when the search box is empty sometimes fails to show all titles

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#1083]
I believe this is due to a protection feature in the search, which prevents lookup if the same string is entered twice in a row (instead, it should just display the previous search). However, there is no previous search if the field is blank and the user has just loaded a ZIM.

I believe the solution is to exempt comparison of current search and prefix if the search box is empty / blank / null, and/or checking that we have search results to display before invoking the protection.

**Comment by Jaifroid** [comment]
@dan-niles Thank you for addressing this issue! I'm assigning you.

**Comment by Jaifroid** [comment]
I'm not sure why, but I can't seem to assign you. Anyway, this is confirmation that you are assigned.

**Comment by Jaifroid** [comment]
@dan-niles Thank you for that PR. Just a word of friendly advice: before undertaking another PR (anywhere on Kiwix, not just here), it's always a good idea to ask to be assigned to the corresponding issue first. The reason is that there are some issues that are out-of-date or no longer wanted, but that we haven't closed. That's not the case for any of the PRs you've done here, I want to stress, but there is that slight risk, and I wouldn't want you to waste your time. Another reason is that if you comment on an issue, I think it is easier for the Repo maintainer to assign you to that issue. Thanks once again for you much-appreciated work.

**Comment by dan-niles** [comment]
@Jaifroid Got it, thanks for the heads up! I'll be sure to ask to be assigned to corresponding issues before starting any future PRs. 👍
